kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java

```diff
@@ -240,6 +241,8 @@ static String getColumnTypeName(TTypeId type) throws SQLException {
         return "struct";
       case NULL_TYPE:
         return "void";
+      case USER_DEFINED_TYPE:
+        return "user_defined";
```

Comment:
Just a question, comply with hive?

Reply:
https://github.com/apache/hive/blob/6dc073b673ec82900424654e5e148625020acd6d/serde/src/java/org/apache/hadoop/hive/serde2/thrift/Type.java#L107C1-L110C20

Yes，it is compatible, just lower case.

Reply:
```
  USER_DEFINED_TYPE("USER_DEFINED",
      java.sql.Types.OTHER,
      TTypeId.USER_DEFINED_TYPE,
      true, false);

  private final String name;
  private final TTypeId tType;
  private final int javaSQLType;
  private final boolean isQualified;
  private final boolean isComplex;
  private final boolean isCollection;

  Type(String name, int javaSQLType, TTypeId tType, boolean isQualified, boolean isComplex,
      boolean isCollection) {
    this.name = name;
    this.javaSQLType = javaSQLType;
    this.tType = tType;
    this.isQualified = isQualified;
    this.isComplex = isComplex;
    this.isCollection = isCollection;
  }
```

Reply:
Also cc @jiaoqingbo related pr: https://github.com/apache/kyuubi/pull/2878